src/Psalm/Internal/PhpVisitor/Reflector/ExpressionResolver.php

```diff
@@ -514,6 +517,19 @@ private static function functionEvaluatesToTrue(
             return false;
         }
 
+        if ($function->name->parts === ['defined']
+            && isset($function->getArgs()[0])
+            && $function->getArgs()[0]->value instanceof PhpParser\Node\Scalar\String_
+        ) {
+            $predefined_constants = get_defined_constants(true);
```

Comment:
Not sure, where / in what context the `ExpressionResolver` is used. Maybe the stubbed constants in `$codebase` should be checked, too. Maybe we should mock `PHP_VERSION` constants to match the values expected for the requested analysis php version.

@orklah What do you think?

Reply:
I'm not sure either :(